### PR TITLE
Compatibility with new Minikore

### PR DESCRIPTION
**Pull request by msaxena2**

### CODE REVIEW

**Comment by msaxena2** [comment]
[Update K to pass with Minikore](https://trello.com/c/8Q4gLrQV/9-update-k-to-pass-with-minikore)

File: java-backend/src/main/java/org/kframework/backend/java/kil/Definition.java [hunk 1]
```diff
@@ -153,7 +154,7 @@ public Definition(MiniKoreUtils.ModuleUtils moduleUtils, KExceptionManager kem)
         JavaConversions.mapAsJavaMap(moduleUtils.signatureFor()).entrySet().stream().forEach(e -> {
             JavaConversions.setAsJavaSet(e.getValue()).stream().forEach(p -> {
                 ImmutableList.Builder<Sort> sortsBuilder = ImmutableList.builder();
-                stream(p._1()).map(s -> Sort.of(s)).forEach(sortsBuilder::add);
+                stream(p._1()).map(s -> Sort.of(s.str())).forEach(sortsBuilder::add);
```
Comment (daejunpark) [inline java-backend/src/main/java/org/kframework/backend/java/kil/Definition.java:157]: Maybe better to have `name()`.

File: java-backend/src/main/scala/RewriterUtils.scala [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 object RewriterUtils {
 
+  val b: Builders = DefaultBuilders
+
   def toLeft(p: Pattern): Pattern = {
     val decodedTuple: (Pattern, Seq[Pattern]) = MiniKoreUtils.decodePatternAttributes(p)
     val lhs = decodedTuple._1 match {
       case Rewrite(left: Pattern, _) => left
-      case Application(symbol: String, items: Seq[Pattern]) => Application(symbol, items map toLeft)
+      case Application(symbol, items: Seq[Pattern]) => b.Application(symbol, items map toLeft)
```
Comment (daejunpark) [inline java-backend/src/main/scala/RewriterUtils.scala:17]: `symbol:Symbol`

File: java-backend/src/main/scala/RewriterUtils.scala [hunk 3]
```diff
@@ -20,7 +24,7 @@ object RewriterUtils {
     val decodedTuple: (Pattern, Seq[Pattern]) = MiniKoreUtils.decodePatternAttributes(p)
     val rhs = decodedTuple._1 match {
       case Rewrite(_ , right: Pattern) => right
-      case Application(symbol: String, items: Seq[Pattern]) => Application(symbol, items map toRight)
+      case Application(symbol, items: Seq[Pattern]) => b.Application(symbol, items map toRight)
```
Comment (daejunpark) [inline java-backend/src/main/scala/RewriterUtils.scala:27]: `symbol:Symbol`

**Review by daejunpark — APPROVED** [review]

**Pull request merged**